astropy/io/ascii/core.py

```diff
@@ -36,6 +36,104 @@
 # Similar dictionary for fast readers
 FAST_CLASSES = {}
 
+
+class CsvWriter(object):
```

Comment:
As far as I can tell the `Dialect` class is just a minor convenience for collecting up all the format parameters into a single object.  It doesn't let you modify the internals of how things are done (which I think is buried in C code).  We need to modify the csv writer to emit `""` (two of the quotechar characters) instead of an empty string for the data in a space-delimited output when the input is an empty string.